libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowManager.m

```diff
@@ -0,0 +1,301 @@
+    BOOL result = NO;
+    if (![self isReservedName:windowName]) {
+        [self.namedWindows removeObjectForKey:windowName];
+        result = YES;
+    }
+    return result;
+}
+
+- (SFSDKWindowContainer *)windowWithName:(NSString *)name {
+    return [self.namedWindows objectForKey:name];
+}
+
+- (void)addDelegate:(id<SFSDKWindowManagerDelegate>)delegate
+{
+    @synchronized (self) {
+        [_delegates addObject:[NSValue valueWithNonretainedObject:delegate]];
+    }
+}
+
+- (void)removeDelegate:(id<SFSDKWindowManagerDelegate>)delegate
+{
+    @synchronized (self) {
+        [_delegates removeObject:[NSValue valueWithNonretainedObject:delegate]];
+    }
+}
+
+- (void)enumerateDelegates:(void (^)(id<SFSDKWindowManagerDelegate> delegate))block
+{
+    @synchronized(self) {
+        [_delegates.allObjects enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+            id<SFSDKWindowManagerDelegate> delegate = [obj nonretainedObjectValue];
+            if (delegate) {
+                if (block) block(delegate);
+            }
+        }];
+    }
+}
+#pragma mark - SFSDKWindowContainerDelegate
+- (void)windowEnable:(SFSDKWindowContainer *_Nonnull)window animated:(BOOL)animated withCompletion:(void (^)(void))completion{
+
+    [self enumerateDelegates:^(id<SFSDKWindowManagerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowManager:willEnableWindow:)]){
+            [delegate windowManager:self willEnableWindow:window];
+        }
+    }];
+    
+    if (animated) {
+        __weak typeof (self) weakSelf = self;
+        UIViewPropertyAnimator *animator = [[UIViewPropertyAnimator alloc] initWithDuration:0.25 curve:UIViewAnimationCurveEaseInOut animations:^{
+            [weakSelf makeOpaqueWithCompletion:window completion:completion];
+        }];
+        [animator startAnimation];
+    } else {
+        [self makeOpaqueWithCompletion:window completion:completion];
+    }
+}
+
+- (void)windowDisable:(SFSDKWindowContainer *)window animated:(BOOL)animated withCompletion:(void (^)(void))completion{
+    [self enumerateDelegates:^(id<SFSDKWindowManagerDelegate> delegate) {
+        if ([delegate respondsToSelector:@selector(windowManager:willDisableWindow:)]){
+            [delegate windowManager:self willDisableWindow:window];
+        }
+    }];
+   
+    if (animated) {
+        __weak typeof (self) weakSelf = self;
+        UIViewPropertyAnimator *animator = [[UIViewPropertyAnimator alloc] initWithDuration:0.25 curve:UIViewAnimationCurveEaseInOut animations:^{
+            [weakSelf makeTransparentWithCompletion:window completion:completion];
+        }];
+        
+        [animator startAnimation];
+    } else {
+        [self makeTransparentWithCompletion:window completion:completion];
+    }
+    
+}
+
+
+#pragma mark - private methods
+- (void)makeTransparentWithCompletion:(SFSDKWindowContainer *)window completion:(void (^)(void))completion {
+    window.window.alpha = 0.0; //make Transparent
```

Comment:
The original alpha value is not remembered. What if the app created a semi-transparent window? Instead we want to show one of our windows, we should make everything else transparent and remember what the key window was and what all the alpha values were. And then we we are done, we should revert the changes and get back to the state the application was in.